### PR TITLE
Tob fixes

### PR DESCRIPTION
**Pull request by kstdl**

### CODE REVIEW

File: src/connectors/NativeConnectorLogic.sol [hunk 1]
```diff
@@ -49,7 +49,16 @@ abstract contract NativeConnectorLogic is TokenConnectorLogic {
         if (lockingValue == 0) {
             revert ZeroValueCall();
         }
-        state.addAmount(msg.sender, lockingValue);
-        emit Locked(msg.sender, lockingValue);
+        if (lockingValue < amount) {
+            revert InsufficientFunds(amount, lockingValue);
+        } else if (lockingValue > amount) {
+            (bool success,) = msg.sender.call{value: lockingValue - amount}("");
```
Comment (VargaElod23) [inline src/connectors/NativeConnectorLogic.sol:55]: Move `state.addAmount(msg.sender, amount)` here before the action happens to be reentrancy-safe.This is not really a problem, but just to be 100% safe

File: src/eth/TaraClient.sol [hunk 2]
```diff
@@ -130,6 +137,16 @@ contract TaraClient is IBridgeLightClient, OwnableUpgradeable, UUPSUpgradeable {
     {
         uint256 signaturesLength = signatures.length;
         for (uint256 i = 0; i < signaturesLength; i++) {
+            if (i > 0) {
+                if (uint256(signatures[i - 1].r) == uint256(signatures[i].r)) {
```
Comment (VargaElod23) [inline src/eth/TaraClient.sol:141]: Could be simplified to 
```
if (uint256(signatures[i - 1].r) == uint256(signatures[i].r) && signatures[i - 1].vs == signatures[i].vs) {
    revert DuplicateSignatures(ECDSA.recover(h, signatures[i].r, signatures[i].vs));
} else if (uint256(signatures[i - 1].r) < uint256(signatures[i].r)) {
    revert SignaturesNotSorted();
}
```

File: src/lib/BridgeBase.sol [hunk 3]
```diff
@@ -187,6 +179,8 @@ abstract contract BridgeBase is Receiver, OwnableUpgradeable, UUPSUpgradeable {
         if (state_with_proof.state.epoch != appliedEpoch + 1) {
             revert NotSuccessiveEpochs({epoch: appliedEpoch, nextEpoch: state_with_proof.state.epoch});
         }
+        // increment applied epoch before applying the state to avoid reentrancy
+        ++appliedEpoch;
```
Comment (VargaElod23) [inline src/lib/BridgeBase.sol:183]: So here if we do this before any action, the function is by itself reentrancy safe. However, it doesn't hurt.

File: test/TaraClient.t.sol [hunk 4]
```diff
@@ -82,6 +82,18 @@ contract TaraClientTest is Test {
         return CompactSignature(r, (bytes32(uint256(v)) << 255) | s);
     }
 
+    function sortSignatures(CompactSignature[] memory signatures) public pure returns (CompactSignature[] memory) {
```
Comment (VargaElod23) [inline test/TaraClient.t.sol:85]: Hmm maybe it is a good idea to put this method as a pure one on the contract itself that takes the signatures? Maybe someone will use it.

**Review by VargaElod23 — CHANGES_REQUESTED** [review]
Small improvements can be done

File: test/TaraClient.t.sol (hunk 4 above)
Comment (kstdl) [reply test/TaraClient.t.sol:85]: I don't see why anyone will need this. It is not hard to sort it in the script

File: src/lib/BridgeBase.sol (hunk 3 above)
Comment (kstdl) [reply src/lib/BridgeBase.sol:180]: Removed reentrancyGuard. Please check where should we also modify the code to make it reentrancy safe or where guard should be added

**Review by rattrap — APPROVED** [review]

**Pull request merged**